manual/zwim.1
.\" generated with Ronn-NG/v0.10.1
.\" http://github.com/apjanke/ronn-ng/tree/0.10.1
.TH "ZWIM" "1" "2024-10-16" "Luis David Licea Torres" "Zwim Manual"
.SH "NAME"
\fBzwim\fR \- a command\-line dictionary based on zim\-tools and w3m
.SH "SYNOPSIS"
\fBzwim\fR [options] [command]
.SH "DESCRIPTION"
Zwim is a flexible, terminal\-based dictionary that provides commands for downloading, searching, and accessing Zim\-format files\.
.SH "COMMANDS"
\fBalter|a\fR \fIlanguage words\|\.\|\.\|\.\fR
.IP "" 4
Remove unwanted language sections and translations from the dictionary entry before viewing it\.
.IP "" 0
.P
\fBalter\-all|aa\fR \fIwords\|\.\|\.\|\.\fR
.IP "" 4
Similar to the \fBalter\fR command, but one tab is opened for every dictionary result\.
.IP "" 0
.P
\fBcopy\-config|c\fR
.IP "" 4
Copy the default configuration file to your Custom Configuration \fI~/\.config/zwim/zwim\.mjs\fR file\. If the \fBXDG_CONFIG_HOME\fR environment variable is set, then the configuration file will be copied to the XDG Configuration \fI$XDG_CONFIG_HOME/zwim/zwim\.mjs\fR file\.
.IP "" 0
.P
\fBdictionary\-download|dd\fR \fIurls\|\.\|\.\|\.\fR
.IP "" 4
Download dictionaries from the given URLs\. If a download is interrupted with \fBctrl+c\fR or otherwise, the partially\-downloaded file will resume to download by executing the command again\.
.IP "" 0
.P
\fBdictionary\-search|ds\fR [\fIlanguages\|\.\|\.\|\.\fR]
.IP "" 4
Show dictionary download URLs associated to the given languages\. The dictionaries can be downloaded by passing their URLs to the \fBdictionary\-download\fR command\. If no languages are given, all the available languages are shown\.
.IP "" 0
.P
\fBfind\-config|f\fR
.IP "" 4
Return the path to the configuration file being currently used\.
.IP "" 0
.P
\fBoutput|o\fR \fIpath language words\|\.\|\.\|\.\fR
.IP "" 4
Output the search results into the given path\.
.IP "" 0
.P
\fBoutput\-alter|oa\fR \fIpath language words\|\.\|\.\|\.\fR
.IP "" 4
Output the pre\-processed search results into the given path\.
.IP "" 0
.P
\fBsearch|s\fR \fIlanguage words\|\.\|\.\|\.\fR
.IP "" 4
Search the dictionary associated to the given language and show dictionary entries based on similarity\.
.IP "" 0
.P
\fBview|v\fR \fIlanguage words\|\.\|\.\|\.\fR
.IP "" 4
View the dictionary entry for the given language and phrase\.
.IP "" 0
.P
\fBview\-all|va\fR \fIwords\.\.\fR
.IP "" 4
Similar to the \fBview\fR command, but one tab is opened for every dictionary result\.
.IP "" 0
.P
\fBhelp\fR [\fIcommand\fR]
.IP "" 4
View the help message for the given command\.
.IP "" 0
.SH "OPTIONS"
\fB\-\-version|\-V\fR
.IP "" 4
Show the program version\.
.IP "" 0
.P
\fB\-\-help|\-h\fR
.IP "" 4
Show a help message\.
.IP "" 0
.SH "NOTES"
\fBAdding Custom Dictionaries and Files\fR
.IP "" 4
To add support for your own Zim\-format dictionaries, go to the configuration file specified by \fBzwim find\-config\fR and add the paths to your files\.
.IP "" 0
.SH "EXAMPLES"
\fBzwim\fR view en a lot
.IP "" 4
View English entry for \fIa lot\fR\.
.IP "" 0
.P
\fBzwim\fR view en dog and \fB<tab>\fR
.IP "" 4
Show suggestions for \fIdog and \|\.\|\.\|\.\fR by pressing \fB<tab>\fR\.
.IP "" 0
.P
\fBzwim\fR view ru quick brown fox jumps over the lazy dog
.IP "" 4
View Russian entry for \fIquick brown fox jumps over the lazy dog\fR\.
.IP "" 0
.P
\fBzwim\fR dictionary\-search english
.IP "" 4
Search for URLs from which to download English dictionaries\.
.IP "" 0
.P
\fBzwim\fR dictionary\-download https://dumps\.wikimedia\.org/other/kiwix/zim/wiktionary/wiktionary_en_simple_all_maxi_2024\-05\.zim
.IP "" 4
Download or resume the download of an English dictionary\.
.IP "" 0
.P
\fBzwim\fR output /tmp/hello\.html en hello
.IP "" 4
Save the English search result of "hello" as an HTML file in the \fI/tmp\fR folder\.
.IP "" 0
.SH "FILES"
Custom Configuration \fI~/\.config/zwim/zwim\.mjs\fR
.IP "" 4
The location of the user's configuration file if the \fBXDG_CONFIG_HOME\fR environment variable is not set\.
.IP "" 0
.P
XDG Configuration \fI$XDG_CONFIG_HOME/zwim/zwim\.mjs\fR
.IP "" 4
The location of the user's configuration file if the \fBXDG_CONFIG_HOME\fR environment variable is set\.
.IP "" 0
.P
Default Configuration \fI/usr/share/zwim/zwim\.mjs\fR
.IP "" 4
The default configuration file\. Instead of modifying this file you should create your custom configuration file by running \fBzwim copy\-config\fR\.
.IP "" 0
.SH "SEE ALSO"
\fBw3m\fR(1), \fBzim\fR(1), \fBzimdump\fR(1), \fBzimsearch\fR(1)


configuration/zwim.mjs
import process from 'node:process';
// Import environment variables to use in paths.
const { HOME, XDG_CACHE_HOME, XDG_DATA_HOME } = process.env;

// The directory where dictionaries will be downloaded.
// You never want to delete this folder.
// You could change it to ${HOME}/Documents/Zwim, for example.
export const dataHome = `${XDG_DATA_HOME ?? `${HOME}/.local/share`}/zwim`;

// The directory where temporary files will be cached.
// You are free to delete this folder.
export const cacheHome = `${XDG_CACHE_HOME ?? `${HOME}/.cache`}/zwim`;

// The dictionaries for "view", "alter", "save", "alter-save", and "search".
// Files will be looked for in the downloadDirectory.
export const dictionaries = {
    // English
    en: /wiktionary_en.*\.zim/,
    // Spanish
    es: /wiktionary_es.*\.zim/,
    // Japanese
    ja: /wiktionary_ja.*\.zim/,
    // Russian
    ru: /wiktionary_ru.*\.zim/,
};

// The dictionaries to include in "view-all" and "alter-all" results.
// The entries correspond to dictionary names.
export const relevantDictionaries = [
    'en',
    'ru',
];

// The translations to keep in "alter" and "alter-all" results.
//
// Language codes: https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes
export const relevantTranslations = [
    'cu', // Church Slavonic
    'el', // Greek
    'en', // English
    'eo', // Esperanto
    'es', // Spanish
    'ja', // Japanese
    'la', // Latin
    'ru', // Russian
];

// The URL from which the dictionaries will be downloaded.
// You can download more dictionaries from these websites:
// https://dumps.wikimedia.org/other/kiwix/zim/wiktionary/
// https://ftp.fau.de/kiwix/zim/wiktionary/
// https://ftp.nluug.nl/kiwix/zim/wiktionary/
export const wiktionaryUrl = 'https://dumps.wikimedia.org/other/kiwix/zim/wiktionary/';
